### PR TITLE
Stabilize unsafe extern blocks

### PR DESCRIPTION
**Pull request by spastorino**
This goes after this is merged:

- https://github.com/rust-lang/rust/pull/127921

...and CI is going to be happy after it.

r? @traviscross 

Tracking:

- https://github.com/rust-lang/rust/issues/123743

### CODE REVIEW

**Review by traviscross — CHANGES_REQUESTED** [review]
There's a TODO in the page for the version of Rust in which we stabilize this.  We'll need to remember to fill that in after it stabilizes and before merging this.

Otherwise this is ready to merge once...

- https://github.com/rust-lang/rust/pull/127921

... is merged.

**Comment by spastorino** [comment]
@traviscross this is now ready as https://github.com/rust-lang/rust/pull/127921 is merged. I've retriggered CI tests and CI is happy now.

**Pull request merged**
